### PR TITLE
Add ln_determinant to Cholesky

### PR DESCRIPTION
**Pull request by Shatur**
This PR implements a function that returns natural logarithm of determinant similar to the one in [NumPy](https://numpy.org/doc/stable/reference/generated/numpy.linalg.slogdet.html) and in [ndarray-linalg](https://docs.rs/ndarray-linalg/latest/ndarray_linalg/cholesky/trait.DeterminantC.html#tymethod.ln_detc).

### CODE REVIEW

File: src/linalg/cholesky.rs [hunk 1]
```diff
@@ -41,6 +42,7 @@ where
 impl<T: SimdComplexField, D: Dim> Cholesky<T, D>
 where
     DefaultAllocator: Allocator<T, D, D>,
+    T::SimdRealField: Zero,
```
Comment (sebcrozet) [inline src/linalg/cholesky.rs:45]: This shouldn’t be needed.

File: src/linalg/cholesky.rs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 use serde::{Deserialize, Serialize};
 
 use num::One;
+use num::Zero;
```
Comment (sebcrozet) [inline src/linalg/cholesky.rs:5]:
```suggestion
use num::{One, Zero};
```

**Review by sebcrozet — CHANGES_REQUESTED** [review]
Thank you for this PR!

File: src/linalg/cholesky.rs (hunk 1 above)
Comment (Shatur) [reply src/linalg/cholesky.rs:45]: But I need to create `T::SimdRealField::zero()`. It won't work with `T::zero()`.

Comment (sebcrozet) [reply src/linalg/cholesky.rs:44]: You can call `T::SimdRealField::zero()` without the explicit `T::SimdRealField: Zero` bound.

Comment (Shatur) [reply src/linalg/cholesky.rs:44]: Oh, right! For some reason I thought that it is necessary.

**Review by sebcrozet — APPROVED** [review]

**Comment by sebcrozet** [comment]
All good, thanks!

**Pull request merged**